clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -372,9 +372,11 @@ Address CIRGenFunction::getAddrOfBitFieldStorage(LValue base,
                                                  unsigned index) {
   mlir::Location loc = getLoc(field->getLocation());
   cir::PointerType fieldPtr = cir::PointerType::get(fieldType);
+  auto rec = cast<cir::RecordType>(base.getAddress().getElementType());
+  if (index == 0 && rec.isUnion())
```

Comment:
Is index ever non-zero for unions? I see that the incubator only checks `if (index == 0)` here and returns `base.getAddress()` in that case.